multicluster/test/e2e/framework.go

```diff
@@ -322,6 +322,100 @@ func createPod(client kubernetes.Interface, name string, namespace string, ctrNa
 	return err
 }
 
+func (data *TestData) probe(
+	cluster string,
+	podNamespace string,
+	podName string,
+	containerName string,
+	dstAddr string,
+	dstName string,
+	port int32,
+	protocol corev1.Protocol,
+) antreae2e.PodConnectivityMark {
+	protocolStr := map[corev1.Protocol]string{
+		corev1.ProtocolTCP:  "tcp",
+		corev1.ProtocolUDP:  "udp",
+		corev1.ProtocolSCTP: "sctp",
+	}
+	// There seems to be an issue when running Antrea in Kind where tunnel traffic is dropped at
+	// first. This leads to the first test being run consistently failing. To avoid this issue
+	// until it is resolved, we try to connect 3 times.
+	// See https://github.com/antrea-io/antrea/issues/467.
+	cmd := []string{
+		"/bin/sh",
+		"-c",
+		fmt.Sprintf("for i in $(seq 1 3); do /agnhost connect %s:%d --timeout=1s --protocol=%s; done;", dstAddr, port, protocolStr[protocol]),
+	}
+	log.Tracef("Running: kubectl exec %s -c %s -n %s -- %s", podName, containerName, podNamespace, strings.Join(cmd, " "))
+	stdout, stderr, err := data.runCommandFromPod(cluster, podNamespace, podName, containerName, cmd)
+	if err != nil {
+		// log this error as trace since may be an expected failure
+		log.Tracef("%s -> %s: error when running command: err - %v /// stdout - %s /// stderr - %s", podName, dstName, err, stdout, stderr)
+		// If err != nil and stderr == "", then it means this probe failed because of
+		// the command instead of connectivity. For example, container name doesn't exist.
+		if stderr == "" {
+			return antreae2e.Error
+		}
+		return decideProbeResult(stderr, 3)
+	}
+	return antreae2e.Connected
+}
+
+// decideProbeResult uses the probe stderr to decide the connectivity.
+func decideProbeResult(stderr string, probeNum int) antreae2e.PodConnectivityMark {
+	countConnected := probeNum - strings.Count(stderr, "\n")
+	countDropped := strings.Count(stderr, "TIMEOUT")
+	// For our UDP rejection cases, agnhost will return:
```

Comment:
I am wondering is OK to test TCP only Service? we only create and export a nginx service with TCP port. or there is actually no difference?

Reply:
I think we should be okay there. ANP on UDP / SCTP traffic is tested extensively in non-multicluster e2e testcases. The major thing to be tested here IMO is that if a policy can be correctly applied to a Service clusterIP exported from another cluster